architecture/2024-11-authz-feature-spec.md

```diff
@@ -0,0 +1,392 @@
+# Radius Authorization Feature Specification
+
+* **Author**: Zach Casper (@zachcasper)
+
+## Summary
+
+Radius is inherently a multi-tenant system with several different personas. Separation of concerns between developers and platform engineers is a core concept. Developers model their applications using resource types abstracted away from the infrastructure while platform engineers configure how those resource types are deployed. Radius is also designed to support multiple different applications, developers, and groups of developers. 
+
+The building blocks of this multi-tenancy is implemented today with resource groups. Every resource, whether it is an application, an environment, or a recipe, resides in a resource group. However, today, access to Radius is binary—you ether have access to the API or you do not. It is not possible to control permissions to these resources and resource groups. This feature specification describes the requirements and user experience for authorizing users to perform a granular set of actions on a granular set of resources and resource groups.
+
+### Goals
+
+This feature specification describes:
+
+* Granting and revoking permission to users to perform Radius administrative actions at the control-plane level and tenant level
```

Comment:
Permission to install, delete and update Radius installation?

Reply:
No, these are external to Radius actions. In these cases, you would need access to the Kubernetes cluster.